shell/app/modules/project/common/components/issue/subscribers-selector.tsx

```diff
@@ -0,0 +1,195 @@
+  issueType: string;
+  projectId: string;
+  data: object;
+  setData: (data: object) => void;
+}
+
+export const SubscribersSelector = (props: IProps) => {
+  const { subscribers: subscribersProps, issueID, issueType, projectId, setData, data } = props;
+  const { id: loginUserId, ...loginUser } = userStore.getState((s) => s.loginUser);
+  const { subscribe, unsubscribe, getIssueDetail, getIssueStreams, batchSubscribe } = issueStore.effects;
+  const memberRef = React.useRef<{ [p: string]: Function }>(null);
+  const [subscribers, setSubscribers] = React.useState<(string | number)[]>([]);
+  const [usersMap, setUsersMap] = React.useState({});
+
+  let isFollowed = subscribers.includes(loginUserId);
+  const userMap = useUserMap();
+
+  React.useEffect(() => {
+    const _userMap: object = { ...userMap };
+    _userMap[loginUserId] || (_userMap[loginUserId] = { ...loginUser, userId: loginUserId });
+    setUsersMap({ ..._userMap });
+  }, []);
+
+  React.useEffect(() => {
+    setSubscribers(subscribersProps || []);
+  }, [subscribersProps]);
+
+  const menu = (
+    <Menu>
+      <Menu.Item>
+        {isFollowed ? (
+          <div
+            onClick={async () => {
+              if (issueID) {
+                await unsubscribe({ id: issueID });
+                getIssueDetail({ id: issueID, type: issueType });
+                getIssueStreams({ type: issueType, id: issueID, pageNo: 1, pageSize: 50 });
+              } else {
+                let index = subscribers.findIndex((item) => item === loginUserId);
+                subscribers.splice(index, 1);
+                setSubscribers([...subscribers]);
+              }
+            }}
+          >
+            <IconPreviewCloseOne />
+            取消关注
+          </div>
+        ) : (
+          <div
+            onClick={async () => {
+              if (issueID) {
+                await subscribe({ id: issueID });
+                getIssueDetail({ id: issueID, type: issueType });
+                getIssueStreams({ type: issueType, id: issueID, pageNo: 1, pageSize: 50 });
+              } else {
+                setSubscribers([...subscribers, loginUserId]);
+              }
+            }}
+          >
+            <IconPreviewOpen />
+            关注
+          </div>
+        )}
+      </Menu.Item>
+      <Menu.Item>
+        <MemberSelector
+          scopeType="project"
+          className="issue-member-select"
+          dropdownClassName="issue-member-select-dropdown"
+          scopeId={projectId}
+          allowClear={false}
+          ref={memberRef}
+          mode="multiple"
+          value={subscribers || []}
+          onChange={async (ids: Array<number | string>, options: Array<{ userId: string }>) => {
+            if (issueID) {
+              await batchSubscribe({ id: issueID, subscribers: ids });
+              getIssueDetail({ id: issueID, type: issueType });
+              getIssueStreams({ type: issueType, id: issueID, pageNo: 1, pageSize: 50 });
```

Comment:
Every time change selected members will recall these apis?

Reply:
better to add a new props `onDropdownVisibleChange` in LoadMoreSelector, and then use like this
```
onDropdownVisibleChange={(vis) => {
    if(!vis && compare value){ // only when vis===false and the value has changed
    }
}}
```